web/ui/react-app/src/pages/rules/Rules.tsx

```diff
@@ -1,15 +1,147 @@
 import PathPrefixProps from '../../types/PathPrefixProps';
-import { Alert } from 'reactstrap';
-
-const Rules: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Rules</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/rules`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+import { Alert, Table } from 'reactstrap';
+import { useFetch } from '../../hooks/useFetch';
+import { Link } from '@reach/router';
+import { Rule } from '../../types/types';
+import { formatRelative, roundUp } from '../../utils';
+import { now } from 'moment';
+
+interface RuleGroup {
+  name: string;
+  file: string;
+  rules: Rule[];
+  evaluationTime: string;
+  lastEvaluation: string;
+}
+
+interface RulesMap {
+  groups: RuleGroup[];
+}
+
+interface MapProps {
+  map: Record<string, string>;
+  term: string;
+}
+
+const AlertRuleMaps: FC<RouteComponentProps & MapProps> = ({ term, map }) => {
+  return (
+    <div>
+      <strong>{term}:</strong>
+      <div style={{ marginLeft: '3%' }}>
+        {Object.entries(map).map(([key, value], i) => (
+          <div key={i}>
+            {key}: {value}
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+};
+
+const Rules: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
+  const { response, error } = useFetch<RulesMap>(`${pathPrefix}/api/v1/rules`);
+
+  if (error) {
+    return (
+      <Alert color="danger">
+        <strong>Error:</strong> Error fetching Rules: {error.message}
+      </Alert>
+    );
+  } else if (response.data) {
+    const groups: RuleGroup[] = response.data.groups;
+    return (
+      <>
+        <h2>Rules</h2>
+        {groups.map((g, i) => {
+          return (
+            <Table size="sm" striped bordered key={i}>
+              <thead>
+                <tr>
+                  <td colSpan={3}>
+                    <a href={'#' + g.name}>
+                      <h2>{g.name}</h2>
+                    </a>
+                  </td>
+                  <td>
+                    <h2>{formatRelative(g.lastEvaluation, now())} ago</h2>
+                  </td>
+                  <td>
+                    <h2>{roundUp(parseFloat(g.evaluationTime) * 1000)}ms</h2>
+                  </td>
+                </tr>
+              </thead>
+              <tbody>
+                <tr className="font-weight-bold">
+                  <td>Rule</td>
+                  <td>State</td>
+                  <td>Error</td>
+                  <td>Last Evaluation</td>
+                  <td>Evaluation Time</td>
+                </tr>
+                {g.rules.map((r, i) => {
+                  return (
+                    <tr key={i}>
+                      {r.alerts ? (
+                        <td>
+                          <strong>alert:</strong>
```

Comment:
I would prefer if we didn't use CSS/HTML spacing (`margin-left: 3%`, `<br/>`, etc.) in this field and instead kept it all preformatted like in the old UI and use existing normal whitespace for the formatting, so it's properly copyable YAML (at least for the most part).

In the old UI the styling was achieved via:

```css
.rule_cell {
    white-space: pre-wrap;
    background-color: #F5F5F5;
    display: block;
    font-family: monospace;
}
```